lib/lambda.js

```diff
@@ -2,6 +2,28 @@
 
 const shimmer = require('./instrumentation/shimmer')
 
+// used to detect cold starts
+let isFirstRun = false
+
+function getTriggerTypeFromEvent (event) {
+  if (event && event.requestContext && event.requestContext.requestId) {
+    return 'http'
+  }
```

Comment:
Do you intend to leave the SNS/SQS/S3-trigger handling to later?

nit: Given that the spec says `faas.trigger.request_id` is dependent on `faas.trigger.type`, I would consider having a single function that handles those two together rather than the pattern currently here which has the `event && event.requestContext && event.requestContext.requestId`-logic in two places.

Reply:
resolving as obsolete -- approach has changed significantly.